### PR TITLE
Create KTooltip component

### PR DESCRIPTION
**Pull request by christianmemije**
### Summary

**TODOS:** 
- [x] Finalize component API
- [x] Stylize. Currently using the default css.
- [x] Use in all places we use the `title` attribute.  
- [x] Add to style guide.
- [ ] Find a workaround for tooltips within containers that haver overflow hidden such as within tables.
- [x] Add tests. 

### Reviewer guidance

Still WIP.

### Contributor Checklist

- [ ] Contributor has fully tested the PR manually
- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

**Comment by indirectlylit** [comment]
refs #4171

File: kolibri/core/assets/src/views/KTooltip.vue [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  <popper
+    :disabled="disabled"
+    :visibleArrow="false"
+    :options="options"
+    trigger="hover"
+    class="container d-ib"
+  >
+    <div
+      slot="reference"
+      class="d-ib"
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/KTooltip.vue]: Curious what the purpose of this is?

It seems potentially brittle for us to be overriding the display style of reference elements.

File: kolibri/core/assets/src/views/KTooltip.vue [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    >
+      <slot name="trigger"></slot>
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/KTooltip.vue]: `trigger` makes me think of an event. maybe `target`? (or just keep `reference` like in popper?)

**Review by indirectlylit — COMMENTED** [review]
I don't particularly like the API exposed by the [`vue-popper` wrapper](https://github.com/RobinCK/vue-popper):

* It uses two named slots, which I find to be less clear than a single default slot or input props
* It wraps the tooltip target (trigger), which adds a layer of DOM nesting that shouldn't need to be there

The API I'd expect would look something more like:

```html
<template>
  <div>
    <UiIcon ref="tooltipTarget">
      <mat-svg name="info" category="action" />
    </UiIcon>
    <KTooltip :target="$refs.tooltipTarget">
      {{ $tr('tooltipText') }}
    </KTooltip>
  </div>
</template>
```

I know it's not typical to pass a `ref` as a prop but I feel like this produces a cleaner DOM structure

File: kolibri/core/assets/src/views/KTooltip.vue [hunk 3]
```diff
@@ -1,23 +1,19 @@
...
-    >
-      <slot name="trigger"></slot>
-    </div>
     <div
       dir="auto"
       class="popper-custom-skin"
     >
-      <slot name="tooltip"></slot>
+      <!--Default slot that will contain the tooltip content.-->
+      <slot></slot>
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/KTooltip.vue:16]: ❤️

File: kolibri/core/assets/src/views/KTooltip.vue [hunk 4]
```diff
@@ -28,16 +24,32 @@
...
     name: 'KTooltip',
     components: {
       Popper,
     },
     props: {
+      /**
+       * Element tooltip will be positioned relative to
+       */
+      reference: {
+        type: HTMLElement,
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/KTooltip.vue]: cool!

**Comment by indirectlylit** [comment]
refs #3723

File: kolibri/core/assets/src/views/KTooltip.vue [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        return this.mounted && this.htmlElement;
+      },
+      options() {
+        return {
+          placement: this.placement,
+          modifiers: {
+            preventOverflow: {
+              // enabled: true,
+              // escapeWithReference: true,
+              // boundariesElement: 'viewport',
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/KTooltip.vue]: should these value be uncommented?

File: kolibri/plugins/style_guide/assets/src/views/shell/VueExample.vue [hunk 6]
```diff
@@ -27,6 +27,6 @@
 
 <style lang="scss" scoped>
 
-  @import '~codemirror/theme/mdn-like.css';
+  @import '~codemirror/theme/dracula.css';
```
Comment (indirectlylit) [inline kolibri/plugins/style_guide/assets/src/views/shell/VueExample.vue:30]: muahaha

**Review by indirectlylit — COMMENTED** [review]
looking good

seeing some weird behavior with text wrapping:

image

**Comment by indirectlylit** [comment]
@christianmemije FYI if you merge in the latest `develop` it will build

File: kolibri/core/assets/src/views/KTooltip.vue (hunk 5 above)
Comment (christianmemije) [reply kolibri/core/assets/src/views/KTooltip.vue]: Removed

File: kolibri/core/assets/src/views/ContentIcon.vue [hunk 7]
```diff
@@ -1,7 +1,7 @@
 <template>
 
-  <span>
-    <UiIcon ref="type-icon">
+  <div>
```
Comment (jonboiser) [inline kolibri/core/assets/src/views/ContentIcon.vue]: Making this a div breaks the layout of rows on the import/export page.

screen shot 2018-11-07 at 5 36 54 pm

File: kolibri/core/assets/test/progress-icon.spec.js [hunk 8]
```diff
@@ -1,51 +1,54 @@
 import { mount } from '@vue/test-utils';
-import UiTooltip from 'keen-ui/src/UiTooltip';
 import UiIcon from 'keen-ui/src/UiIcon';
 import ProgressIcon from '../src/views/ProgressIcon';
 
 function testIcon(wrapper, expectedText) {
   // prettier-ignore
-  expect(wrapper.find(UiTooltip).text().trim()).toEqual(expectedText);
+  expect(wrapper.find({ name: 'KTooltip'}).text().trim()).toEqual(expectedText);
```
Comment (jonboiser) [inline kolibri/core/assets/test/progress-icon.spec.js]: Run the `npm run lint-js:fix` script to auto-format the tests.

File: kolibri/core/assets/src/views/KTooltip.vue [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+    props: {
+      /**
+       * String of ref which tooltip will be positioned relative to.
+       */
+      reference: {
+        type: String,
+        required: true,
+      },
+      /**
+       * Refs object.
```
Comment (jonboiser) [inline kolibri/core/assets/src/views/KTooltip.vue]: Maybe write more here explaining why you need to pass parents' refs into KTooltip

**Review by jonboiser — APPROVED** [review]
Looks good to me!

**Pull request merged**